app/helpers/federation_helper.rb

```diff
@@ -124,4 +124,59 @@ def federation_external_class?(class_object)
   end
 
 
+  def federation_portal_status(portal_name: nil)
+    Rails.cache.fetch("federation_portal_up_#{portal_name}", expires_in: 2.hours) do
+      portal_api = federated_portals&.dig(portal_name,:api)
+      return false unless portal_api
+      portal_up = false
+      begin
+        response = Faraday.new(url: portal_api) do |f|
+          f.adapter Faraday.default_adapter
+          f.request :url_encoded
+          f.options.timeout = 20
+          f.options.open_timeout = 20
+        end.head
+        portal_up = response.success?
+      rescue StandardError => e
+        Rails.logger.error("Error checking portal status for #{portal_name}: #{e.message}")
+      end
+      portal_up
+    end
+  end
+
+  def federation_portals_status(key,name,acronym,checked,portal_up)
```

Comment:
This function and the one in the line 127 have a very close name, we should change the name of this to something like:
`federation_chip_component`

Reply:
ok

Reply:
done in https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/769/commits/321b81747cfebeba66dee98143227ddc4b4d9fae